Data source azurerm_servicebus_queue_authorization_rule fails to repond and crashes

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.1.9

### AzureRM Provider Version

3.3.0

### Affected Resource(s)/Data Source(s)

azurerm_servicebus_queue_authorization_rule

### Terraform Configuration Files

```hcl
locals {
  rg_name     = "shared_rg"
  location    = "norwayeast"
  queue_name  = "notificationmail"
  queue_rule_name = "Send"
  sb_name     = "myservicebusnamespace45321"
  sb_sku      = "Standard"
  sb_capacity = 0
}

#######################################################
######## CREATE RESOURCES

resource "azurerm_resource_group" "rg" {
  name     = local.rg_name
  location = local.location
}

resource "azurerm_servicebus_namespace" "sbns" {
  depends_on = [azurerm_resource_group.rg]

  name                = local.sb_name
  resource_group_name = azurerm_resource_group.rg.name
  location            = azurerm_resource_group.rg.location
  sku                 = local.sb_sku
  capacity            = local.sb_capacity
}
resource "azurerm_servicebus_queue" "queue" {
  depends_on = [azurerm_servicebus_namespace.sbns]

  name                  = var.queue_name
  namespace_id          = azurerm_servicebus_namespace.sbns.id
  resource_group_name   = azurerm_resource_group.rg.name
  lock_duration         = "PT2M"
  max_size_in_megabytes = 1024
  enable_partitioning   = false
}

resource "azurerm_servicebus_namespace_authorization_rule" "rule_send" {
  depends_on = [azurerm_servicebus_namespace.sbns]

  name                = local.queue_name
  namespace_id        = azurerm_servicebus_namespace.sbns.id
  listen              = false
  send                = true
  manage              = false
}

#######################################################
######## Get data

data "azurerm_servicebus_queue_authorization_rule" "send" {
  depends_on = [azurerm_servicebus_namespace_authorization_rule.rule_send]

  name                = local.queue_rule_name
  queue_name          = local.queue_name
  resource_group_name = local.rg_name
  namespace_name      = local.sb_name
}
```


### Debug Output/Panic Output

```shell
╷
│ Error: Plugin did not respond
│
│   with data.azurerm_servicebus_queue_authorization_rule.send,
│   on data-2-sbus.tf line 16, in data "azurerm_servicebus_queue_authorization_rule" "send":
│   16: data "azurerm_servicebus_queue_authorization_rule" "send" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.

Stack trace from the terraform-provider-azurerm_v3.3.0_x5.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 80 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus.dataSourceServiceBusQueueAuthorizationRuleRead(0x19c525ccf50?, {0x51cd1e0?, 0xc00029e700?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/servicebus/servicebus_queue_authorization_rule_data_source.go:134 +0xd0a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x5acd39f?, {0x6204228?, 0xc001f3b600?}, 0xc001e31ce0?, {0x51cd1e0?, 0xc00029e700?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:346 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc001309ea0, {0x6204228, 0xc001f3b600}, 0xc001e2d480, {0x51cd1e0, 0xc00029e700})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:569 +0xf7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0004ecdf8, {0x6204228, 0xc001f3b600}, 0xc001f78d20)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:1133 +0x339
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0001c8680, {0x62042d0?, 0xc001e31350?}, 0xc001e187d0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:478 +0x31b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x571e0c0?, 0xc0001c8680}, {0x62042d0, 0xc001e31350}, 0xc001f60d20, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:416 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000228a80, {0x62152e0, 0xc001262600}, 0xc0021e8900, 0xc00068dc80, 0xa042430, 0x0)
        google.golang.org/grpc@v1.39.0/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000228a80, {0x62152e0, 0xc001262600}, 0xc0021e8900, 0x0)
        google.golang.org/grpc@v1.39.0/server.go:1617 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.39.0/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.39.0/server.go:938 +0x28a

Error: The terraform-provider-azurerm_v3.3.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Expected to retrieve data service bus queue authorization rule 

### Actual Behaviour

It crashes after running the terraform plan. This Bug started after azurerm provider 3..X.X. Data sources for azurerm_servicebus_queue_authorization_rule works <= 2.2.9. I believe this is related to arguments `queue_name` and `queue_id`. The resources changed arguments to queue_id in 3.X.X

### Steps to Reproduce

It crashes after running the terraform plan. This Bug started after azurerm provider 3..X.X. Data sources for azurerm_servicebus_queue_authorization_rule works <= 2.2.9. I believe this is related to arguments `queue_name` and `queue_id`

terraform init
terraform plan => crashes

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/servicebus/servicebus_queue_authorization_rule_data_source.go
+++ b/internal/services/servicebus/servicebus_queue_authorization_rule_data_source.go
@@ -131,7 +131,7 @@ func dataSourceServiceBusQueueAuthorizationRuleRead(d *pluginsdk.ResourceData, m
 	} else {
 		rgName = d.Get("resource_group_name").(string)
 		nsName = d.Get("namespace_name").(string)
-		queueName = d.Get("topic_name").(string)
+		queueName = d.Get("queue_name").(string)
 	}
 
 	id := parse.NewQueueAuthorizationRuleID(subscriptionId, rgName, nsName, queueName, d.Get("name").(string))

--- a/internal/services/servicebus/servicebus_queue_authorization_rule_data_source_test.go
+++ b/internal/services/servicebus/servicebus_queue_authorization_rule_data_source_test.go
@@ -32,6 +32,30 @@ func TestAccDataSourceServiceBusQueueAuthorizationRule_basic(t *testing.T) {
 	})
 }
 
+func TestAccDataSourceServiceBusQueueAuthorizationRule_queueName(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_servicebus_queue_authorization_rule", "test")
+	r := ServiceBusQueueAuthorizationRuleDataSource{}
+
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: r.queueName(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("id").Exists(),
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("resource_group_name").Exists(),
+				check.That(data.ResourceName).Key("namespace_name").Exists(),
+				check.That(data.ResourceName).Key("queue_name").Exists(),
+				check.That(data.ResourceName).Key("primary_key").Exists(),
+				check.That(data.ResourceName).Key("secondary_key").Exists(),
+				check.That(data.ResourceName).Key("primary_connection_string").Exists(),
+				check.That(data.ResourceName).Key("secondary_connection_string").Exists(),
+				check.That(data.ResourceName).Key("primary_connection_string_alias").HasValue(""),
+				check.That(data.ResourceName).Key("secondary_connection_string_alias").HasValue(""),
+			),
+		},
+	})
+}
+
 func TestAccDataSourceServiceBusQueueAuthorizationRule_withAliasConnectionString(t *testing.T) {
 	data := acceptance.BuildTestData(t, "data.azurerm_servicebus_queue_authorization_rule", "test")
 	r := ServiceBusQueueAuthorizationRuleDataSource{}
@@ -57,6 +81,19 @@ func (ServiceBusQueueAuthorizationRuleDataSource) basic(data acceptance.TestData
 data "azurerm_servicebus_queue_authorization_rule" "test" {
   name     = azurerm_servicebus_queue_authorization_rule.test.name
   queue_id = azurerm_servicebus_queue.test.id
+}
+`, ServiceBusQueueAuthorizationRuleResource{}.base(data, true, true, true))
+}
+
+func (ServiceBusQueueAuthorizationRuleDataSource) queueName(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_servicebus_queue_authorization_rule" "test" {
+  name                = azurerm_servicebus_queue_authorization_rule.test.name
+  resource_group_name = azurerm_resource_group.test.name
+  namespace_name      = azurerm_servicebus_namespace.test.name
+  queue_name          = azurerm_servicebus_queue.test.name
 }
 `, ServiceBusQueueAuthorizationRuleResource{}.base(data, true, true, true))
 }

